pkg/scaffold/resource/resource.go

```diff
@@ -79,8 +79,11 @@ func (r *Resource) Validate() error {
 		return fmt.Errorf(
 			"version must match ^v\\d+(alpha\\d+|beta\\d+)?$ (was %s)", r.Version)
 	}
-	if r.Kind != flect.Pascalize(r.Kind) {
-		return fmt.Errorf("kind must be PascalCase (expected %s was %s)", flect.Pascalize(r.Kind), r.Kind)
+
+	validationErrors := isDNS1035Label(strings.ToLower(r.Kind))
```

Comment:
my recommendation here to solve it is:
- Remove the Pascalize
- Check if the Kind matches with: 
`go
	ResourceKindRegexp = regexp.MustCompile("^[A-Z]{1}[a-zA-Z0-9]+$")
`